In [73]:
import numpy as np
from scipy.stats import entropy
import pandas as pd
import math
import matplotlib.pyplot as plt
import wooldridge as woo
import statsmodels.formula.api as smf

# Step 0 : We import the necessary files

In [74]:
dataFF_red_norm01 = pd.read_csv('dataFF_red_norm01.csv')
dataFF_red_norm_gauss = pd.read_csv('dataFF_red_norm_gauss.csv')

# Step 2 : We do regressions

In [75]:
reg = smf.ols(formula='sum~ Nb_doct_8km', data=dataFF_red_norm_gauss)
results = reg.fit()
b = results.params
print(f'b: \n{b}\n')

"""# scatter plot and fitted values:
plt.plot('APL', 'Nb_carac_présentation', data=dataFF_red_norm, color='grey', marker='o', linestyle='')
plt.plot(dataFF_red_norm['APL'], results.fittedvalues, color='black', linestyle='-')
plt.ylabel('Nb_caracteres_presentation')
plt.xlabel('APL')"""

b: 
Intercept     -0.271677
Nb_doct_8km    3.602223
dtype: float64



"# scatter plot and fitted values:\nplt.plot('APL', 'Nb_carac_présentation', data=dataFF_red_norm, color='grey', marker='o', linestyle='')\nplt.plot(dataFF_red_norm['APL'], results.fittedvalues, color='black', linestyle='-')\nplt.ylabel('Nb_caracteres_presentation')\nplt.xlabel('APL')"

In [76]:
# print results using summary:
print(f'results.summary(): \n{results.summary()}\n')

# print regression table:
table = pd.DataFrame({'b': round(results.params, 4),
                      'se': round(results.bse, 4),
                      't': round(results.tvalues, 4),
                      'pval': round(results.pvalues, 4)})
print(f'table: \n{table}\n')

results.summary(): 
                            OLS Regression Results                            
Dep. Variable:                    sum   R-squared:                       0.003
Model:                            OLS   Adj. R-squared:                  0.003
Method:                 Least Squares   F-statistic:                     22.15
Date:                Tue, 22 Oct 2024   Prob (F-statistic):           2.58e-06
Time:                        16:36:16   Log-Likelihood:                -23268.
No. Observations:                6333   AIC:                         4.654e+04
Df Residuals:                    6331   BIC:                         4.655e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
Intercept      -0.2717      0.

In [62]:
reg = smf.ols(formula='sum~ Nb_doct_2km+Nb_doct_8km+Nb_doct_30km+Nb_doct_f_2km+Nb_doct_f_8km+Nb_doct_f_30km', data=dataFF_red_norm_gauss)
results = reg.fit()
b = results.params
print(f'b: \n{b}\n')


b: 
Intercept          -0.275568
Nb_doct_2km      -382.532551
Nb_doct_8km       -69.634074
Nb_doct_30km       -1.580851
Nb_doct_f_2km     693.034708
Nb_doct_f_8km     161.105423
Nb_doct_f_30km      0.886223
dtype: float64



In [63]:
# print results using summary:
print(f'results.summary(): \n{results.summary()}\n')

# print regression table:
table = pd.DataFrame({'b': round(results.params, 4),
                      'se': round(results.bse, 4),
                      't': round(results.tvalues, 4),
                      'pval': round(results.pvalues, 4)})
print(f'table: \n{table}\n')
#vote1.describe()

results.summary(): 
                            OLS Regression Results                            
Dep. Variable:                    sum   R-squared:                       0.012
Model:                            OLS   Adj. R-squared:                  0.011
Method:                 Least Squares   F-statistic:                     12.28
Date:                Tue, 22 Oct 2024   Prob (F-statistic):           8.70e-14
Time:                        16:22:05   Log-Likelihood:                -22753.
No. Observations:                6204   AIC:                         4.552e+04
Df Residuals:                    6197   BIC:                         4.557e+04
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
Intercept         -0.275

## Regression sur l'instrument

In [77]:
dataFF_red_norm01.columns

Index(['Unnamed: 0', 'Nb_Skills', 'Payment_method', 'Social_security_card',
       'site', 'Nb_charac_presentation', 'Nb_languages', 'nb_charac_card_1',
       'Transport', 'Parking', 'Contact', 'Without_appointment',
       'nb_terms_card_0', 'nb_terms_card_3', 'nb_terms_card_4',
       'nb_charac_card_0', 'nb_charac_card_3', 'nb_charac_card_4',
       'National_&_university_degrees', 'Other_formations_b', 'Experiences_b',
       'Formations_b', 'Work_&_publications_b', 'RPPS', 'Nb_price',
       'Min_price', 'Max_price', 'MED14', 'share_F',
       'Population standardisée par la consommation de soins par tranche d'âge',
       'Sector', 'Nb_doct_2km', 'Nb_doct_8km', 'Nb_doct_30km', 'Nb_doct_f_2km',
       'Nb_doct_f_8km', 'Nb_doct_f_30km', 'Nb_doct_D_2km', 'Nb_doct_D_8km',
       'Nb_doct_D_30km', 'Nb_doct_f_D_2km', 'Nb_doct_f_D_8km',
       'Nb_doct_f_D_30km', 'APL', 'Fibre', 'share_Doctolib'],
      dtype='object')

In [78]:
dataFF_red_norm01=dataFF_red_norm01.rename(columns={"Population standardisée par la consommation de soins par tranche d'âge": "Population_standardisee_par_la_consommation_de_soins_par_tranche_d_age"})

In [79]:
reg = smf.ols(formula='share_Doctolib~Fibre+share_F+Population_standardisee_par_la_consommation_de_soins_par_tranche_d_age', data=dataFF_red_norm01)
results = reg.fit()
b = results.params
print(f'b: \n{b}\n')

b: 
Intercept                                                                 4.517291e-01
Fibre                                                                    -2.655086e-01
share_F                                                                  -1.141919e-02
Population_standardisee_par_la_consommation_de_soins_par_tranche_d_age   -8.325882e-07
dtype: float64



In [80]:
# print results using summary:
print(f'results.summary(): \n{results.summary()}\n')

# print regression table:
table = pd.DataFrame({'b': round(results.params, 4),
                      'se': round(results.bse, 4),
                      't': round(results.tvalues, 4),
                      'pval': round(results.pvalues, 4)})
print(f'table: \n{table}\n')

latex_table = table.to_latex(index=True, header=True, column_format='lcccccc', float_format="%.4f")

print(latex_table)

results.summary(): 
                            OLS Regression Results                            
Dep. Variable:         share_Doctolib   R-squared:                       0.145
Model:                            OLS   Adj. R-squared:                  0.145
Method:                 Least Squares   F-statistic:                     357.9
Date:                Tue, 22 Oct 2024   Prob (F-statistic):          1.06e-214
Time:                        16:36:33   Log-Likelihood:                 1660.9
No. Observations:                6333   AIC:                            -3314.
Df Residuals:                    6329   BIC:                            -3287.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                                                                             coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------

In [68]:
# Convertir en DataFrame
summary_df = results.summary2().tables[0]  # La table des coefficients
type(summary_df)
# Générer le LaTeX à partir du DataFrame
latex_table = summary_df.to_latex()

# Afficher la table LaTeX
print(latex_table)


\begin{tabular}{lllll}
\toprule
 & 0 & 1 & 2 & 3 \\
\midrule
0 & Model: & OLS & Adj. R-squared: & 0.153 \\
1 & Dependent Variable: & share_Doctolib & AIC: & -2913.3967 \\
2 & Date: & 2024-10-22 16:22 & BIC: & -2886.4649 \\
3 & No. Observations: & 6204 & Log-Likelihood: & 1460.7 \\
4 & Df Model: & 3 & F-statistic: & 373.5 \\
5 & Df Residuals: & 6200 & Prob (F-statistic): & 5.67e-223 \\
6 & R-squared: & 0.153 & Scale: & 0.036585 \\
\bottomrule
\end{tabular}



the instrument is sufficiently strong/relevant to identify the endogenous variable if the first-stage f-statistic is larger than 10 and here F-statistic = 357.9, the instrument is then relevant

In [81]:
reg = smf.ols(formula='Nb_doct_D_30km~Fibre+MED14+share_F+Population_standardisee_par_la_consommation_de_soins_par_tranche_d_age+APL', data=dataFF_red_norm01)
results = reg.fit()
b = results.params
#print(f'b: \n{b}\n')

# print results using summary:
print(f'results.summary(): \n{results.summary()}\n')

# print regression table:
table = pd.DataFrame({'b': round(results.params, 4),
                      'se': round(results.bse, 4),
                      't': round(results.tvalues, 4),
                      'pval': round(results.pvalues, 4)})
print(f'table: \n{table}\n')


results.summary(): 
                            OLS Regression Results                            
Dep. Variable:         Nb_doct_D_30km   R-squared:                       0.741
Model:                            OLS   Adj. R-squared:                  0.630
Method:                 Least Squares   F-statistic:                     6.679
Date:                Tue, 22 Oct 2024   Prob (F-statistic):               0.00
Time:                        16:37:02   Log-Likelihood:                 21427.
No. Observations:                6333   AIC:                        -3.905e+04
Df Residuals:                    4430   BIC:                        -2.620e+04
Df Model:                        1902                                         
Covariance Type:            nonrobust                                         
                                                                             coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------

In [82]:
reg = smf.ols(formula='Nb_doct_30km~Fibre+share_F+Population_standardisee_par_la_consommation_de_soins_par_tranche_d_age+APL', data=dataFF_red_norm01)
results = reg.fit()
b = results.params
#print(f'b: \n{b}\n')

# print results using summary:
print(f'results.summary(): \n{results.summary()}\n')

# print regression table:
table = pd.DataFrame({'b': round(results.params, 4),
                      'se': round(results.bse, 4),
                      't': round(results.tvalues, 4),
                      'pval': round(results.pvalues, 4)})
print(f'table: \n{table}\n')

results.summary(): 
                            OLS Regression Results                            
Dep. Variable:           Nb_doct_30km   R-squared:                       0.089
Model:                            OLS   Adj. R-squared:                  0.089
Method:                 Least Squares   F-statistic:                     154.9
Date:                Tue, 22 Oct 2024   Prob (F-statistic):          1.22e-126
Time:                        16:37:04   Log-Likelihood:                -158.44
No. Observations:                6333   AIC:                             326.9
Df Residuals:                    6328   BIC:                             360.6
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
                                                                             coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------

In [90]:
dataFF_red_norm01['MED14']=dataFF_red_norm01['MED14'].apply(lambda x : float(x.replace(',','.')))

In [92]:
reg = smf.ols(formula='Nb_doct_D_30km~Fibre+share_F+Population_standardisee_par_la_consommation_de_soins_par_tranche_d_age+APL+MED14', data=dataFF_red_norm01)
results = reg.fit()
b = results.params
#print(f'b: \n{b}\n')

# print results using summary:
print(f'results.summary(): \n{results.summary()}\n')

# print regression table:
table = pd.DataFrame({'b': round(results.params, 4),
                      'se': round(results.bse, 4),
                      't': round(results.tvalues, 4),
                      'pval': round(results.pvalues, 4)})
print(f'table: \n{table}\n')

results.summary(): 
                            OLS Regression Results                            
Dep. Variable:         Nb_doct_D_30km   R-squared:                       0.151
Model:                            OLS   Adj. R-squared:                  0.150
Method:                 Least Squares   F-statistic:                     224.9
Date:                Tue, 22 Oct 2024   Prob (F-statistic):          1.47e-221
Time:                        16:54:26   Log-Likelihood:                 17662.
No. Observations:                6333   AIC:                        -3.531e+04
Df Residuals:                    6327   BIC:                        -3.527e+04
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
                                                                             coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------